sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -353,6 +353,18 @@ public class SentryOptions {
    */
   private boolean enableDeduplication = true;
 
+  /**
+   * Enables event size limiting with {@link EventSizeLimitingEventProcessor}. When enabled, events
+   * exceeding 1MB will have breadcrumbs and stack frames reduced to stay under the limit.
+   */
+  private boolean enableEventSizeLimiting = false;
+
+  /**
+   * Callback invoked when an oversized event is detected. This allows custom handling of oversized
+   * events before the automatic reduction steps are applied.
+   */
+  private @Nullable OnOversizedErrorCallback onOversizedError;
```

Comment:
What should we best call this so customers know what it's used for?
Should this instead be more like an `EventProcessor` interface, where we can add methods for each feature (error events, spans, ...) in the future?
```suggestion
  private @Nullable OnOversizedEventCallback onOversizedEvent;
```